### ISSUE TITLE
Invoke-NullCoalescing lies

### ISSUE THREAD

**Issue by wekempf** [dahlbyk/posh-git#93]
I'm not sure why this command is exported publicly? It has nothing to do with Git, and as such I'd prefer it to not be exported, honestly. If you are going to export it, though, it should do what the name says it does.

PS> Invoke-NullCoalescing '' 'oops'
oops
PS> Invoke-NullCoalescing $false 'oops'
oops
PS> Invoke-NullCoalescing 0 'oops'
oops

I'm sure you can think of many other ways to get the wrong result here. However, there is one more example that you may consider correct, but I find to be very confusing.

PS> Invoke-NullCoalescing { } 'oops'
oops

Invoking the ScriptBlocks may be what you need, but it seems very surprising to me. If I were to implement this it would have been simply:

``` PowerShell
function Invoke-NullCoalescing {
   $args | Where-Object { $_ -ne $null } | Select-Object -First 1
}
```

To include the script block processing, something like:

``` PowerShell
function Invoke-NullCoalescing([switch]$ProcessScriptBlocks) {
   $args | ForEach-Object {
        if ($_ -is [ScriptBlock] -and $ProcessScriptBlocks) {
            & $_
        } else {
            $_
        }
    } | Where-Object { $_ -ne $null } | Select-Object -First 1
}
```

**Comment by kamranayub** [comment]
I just got a new W10 Anniversary image on my PC and tried `Install-Module posh-git` and am now seeing this:

```
PackageManagement\Install-Package : A command with name
'Invoke-NullCoalescing' is already available on this system.
This module 'posh-git' may override the existing commands.
If you still want to install this module 'posh-git', use
-AllowClobber parameter.
At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\1 .0.0.1\PSModule.psm1:1661 char:21
```

Looks like I have this cmdlet already from PSCX that must be installed. I'll keep playing around, but this is the first time I've seen this.

**Comment by jcislowski** [comment]
1. Cannot install posh-git whilst the Powershell Community extensions (pscx) is installed.  
2. posh-git should also not be setting an alias on ??,   
3. should it export this function either since it is an internal function only.  
4. Manifest for posh-git should have -DefaultCommandPrefix set to prevent conflicts with the users session.

**Comment by rkeithhill** [comment]
I believe we are planning to not export this function in the 1.0.0 release.  It will remain in the upcoming 0.7.0 release as we are trying to avoid potentially breaking changes.  The 1.0.0 release however will gives us some latitude WRT breaking changes.

**Comment by rkeithhill** [comment]
Closed with PR #427
